### ISSUE TITLE
Warnings in `hard_case_DMRnet_insurance.R`

### ISSUE THREAD

**Issue by SzymonNowakowski** [SzymonNowakowski/DMRnet#33]
There was a warning
```{r}
Warning message:
In min(sp[[kt]][sp[[kt]] != 1]) :
  no non-missing arguments to min; returning Inf
```
in tests conducted for 0.3.1 version (see the [relevant tag](https://github.com/SzymonNowakowski/DMRnet/releases/tag/v0.3.1_release_candidate_all_tests_passed))

**Comment by SzymonNowakowski** [comment]
Not observed in `hard_case_DMRnet_insurance.R` results for 0.3.2. Maybe a different machine was involved. 

**Action**: Check if it is the computer settings related to warning printout that are involved in printing/not printing this warning.
**Action**: try repeating all `hard_case...` tests on a more conservative machine

**Comment by SzymonNowakowski** [comment]
I rerun the tests for commit `3091ad4` on three machines: `dw`, `dr` and `dg`. The tests on `dw` and `dr` were positive (=reproduced the issue) , while the test on `dg` was not (=not reproduced). It is unclear for me why it is not present on `dg`.

Thus I further worked on `dw`. I was able to reproduce the issue on 0.3.2.9001 version:
```
Loading required package: usethis
ℹ Loading DMRnet
Loaded DMRnet version 0.3.2.9001
Warning messages:
1: In FUN(X[[i]], ...) :
  input string '紐約, NY, United States' cannot be translated to UTF-8, is it valid in 'CP1250'?
2: In FUN(X[[i]], ...) :
  input string 'Бруклин , NY, United States' cannot be translated to UTF-8, is it valid in 'CP1250'?
3: In FUN(X[[i]], ...) :
  input string '布鲁克林, NY, United States' cannot be translated to UTF-8, is it valid in 'CP1250'?
glamer single 2-level factor column:
dmrnet single 2-level factor column:
dmr single 2-level factor column:
glamer:
dmrnet:
cv.dmrnet:
dmr:
cv.dmr:
binomial glamer single 2-level factor column:
binomial dmrnet single 2-level factor column:
binomial dmr single 2-level factor column:
binomial glamer:
binomial dmrnet:
binomial cv.dmrnet:
Warning message:
In min(sp[[kt]][sp[[kt]] != 1]) :
  no non-missing arguments to min; returning Inf
binomial dmr:
binomial cv.dmr:
completed
```

The three warnings just after the package got loaded are related to the `airbnb` dataset present in `testing_branch`, it is not present in `master` branch, so they are not of our concern now. 

What is of our concern is the successful reproduction of 
```
Warning message:
In min(sp[[kt]][sp[[kt]] != 1]) :
  no non-missing arguments to min; returning Inf
```

**Comment by SzymonNowakowski** [comment]
The `min(sp` string can be found in 6 files:
```{bash}
$ grep "min(sp" R/*.R
R/clusters_4glm_help.R:        dod <- min(sp[[kt]][sp[[kt]] != 1])
R/clusters_4glm_help.R:          sp[[kt]][sp[[kt]] != 1] <- sp[[kt]][sp[[kt]] != 1] + dod - min(sp[[kt]][sp[[kt]] != 1])
R/clusters_4lm_help.R:        dod <- min(sp[[kt]][sp[[kt]] != 1])
R/clusters_4lm_help.R:          sp[[kt]][sp[[kt]] != 1] <- sp[[kt]][sp[[kt]] != 1] + dod - min(sp[[kt]][sp[[kt]] != 1])
R/DMR4glm_help.R:           dod <- min(sp[[kt]][sp[[kt]] != 1])
R/DMR4glm_help.R:                                       sp[[kt]][sp[[kt]] != 1] <- sp[[kt]][sp[[kt]] != 1] + dod - min(sp[[kt]][sp[[kt]] != 1])
R/DMR4glm.R:           dod <- min(sp[[kt]][sp[[kt]] != 1])
R/DMR4glm.R:                                       sp[[kt]][sp[[kt]] != 1] <- sp[[kt]][sp[[kt]] != 1] + dod - min(sp[[kt]][sp[[kt]] != 1])
R/DMR4lm_help.R:           dod <- min(sp[[kt]][sp[[kt]] != 1])
R/DMR4lm_help.R:                                       sp[[kt]][sp[[kt]] != 1] <- sp[[kt]][sp[[kt]] != 1] + dod - min(sp[[kt]][sp[[kt]] != 1])
R/DMR4lm.R:           dod <- min(sp[[kt]][sp[[kt]] != 1])
R/DMR4lm.R:                                       sp[[kt]][sp[[kt]] != 1] <- sp[[kt]][sp[[kt]] != 1] + dod - min(sp[[kt]][sp[[kt]] != 1])
```

**Comment by SzymonNowakowski** [comment]
Candidate fix issued in 0.3.2.9004

**Comment by SzymonNowakowski** [comment]
fixed in 0.3.3 - see the [relevant tag](https://github.com/SzymonNowakowski/DMRnet/releases/tag/v0.3.3_release_candidate_all_tests_passed)
